selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/workspace/CheTestWorkspaceProvider.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.selenium.core.workspace;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import com.google.inject.Inject;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import javax.inject.Named;
+import org.eclipse.che.commons.lang.concurrent.LoggingUncaughtExceptionHandler;
+import org.eclipse.che.selenium.core.client.CheTestWorkspaceServiceClient;
+import org.eclipse.che.selenium.core.client.CheTestWorkspaceServiceClientFactory;
+import org.eclipse.che.selenium.core.user.DefaultTestUser;
+import org.eclipse.che.selenium.core.utils.WorkspaceDtoDeserializer;
+
+public class CheTestWorkspaceProvider extends AbstractTestWorkspaceProvider {
+
+  @Inject
+  CheTestWorkspaceProvider(
+      @Named("che.workspace_pool_size") String poolSize,
+      @Named("che.threads") int threads,
+      @Named("workspace.default_memory_gb") int defaultMemoryGb,
+      DefaultTestUser defaultUser,
+      WorkspaceDtoDeserializer workspaceDtoDeserializer,
+      CheTestWorkspaceServiceClient testWorkspaceServiceClient,
```

Comment:
We declare `TestWorkspaceServiceClient` and `TestWorkspaceServiceClientFactory` here in constructor.